src/core/lib/http/httpcli.cc

```diff
@@ -41,17 +41,59 @@
 #include "src/core/lib/iomgr/tcp_client.h"
 #include "src/core/lib/slice/slice_internal.h"
 
-struct internal_request {
+namespace {
+
+struct InternalRequest {
```

Comment:
Let's turn this into a proper class and change the static functions below to be methods.